scipy/stats/_hypotests.py

```diff
@@ -692,23 +692,25 @@ def barnard_exact(table, alternative="two-sided", pooled=True, num_it=3):
         * 'greater': one-sided
 
     pooled : bool, optional
-        Whether to compute score statistic with pooled variance or
-        unpooled (Wald statistic)
+        Whether to compute score statistic with pooled variance (Student's
+        t-test) or unpooled variance (Welch's t-test). Default is ``True`` :
+        Statistic test is computed using pooled variance
 
     num_it : int, optional
         Number of iterations of the grid search. Default is 3. Must be
-        non-negative.
+        non-negative. In most cases, 3 iterations is perfectly enough to
+        reach a precision a goood precision. Above a certain number (around
+        6 iterations), the result will not change anymore. Note that every
+        iteration added comes with a performance cost.
 
     Returns
     -------
-    ber : ``BarnardExactResult`` object
-            The object has attributes ``statistic`` and ``pvalue``.
-            ``statistic`` is the Z statistic with pooled (Score) or
-            unpooled (Wald) variance, depending on the user choice of
-            `pooled` param.
-            ``pvalue`` the probability of obtaining a test statistic at least
-            as extreme as the one that was actually observed, assuming that
-            the null hypothesis is true.
+    ber : ``BarnardExactResult`` object The object has attributes
+          ``statistic`` and ``pvalue``. ``statistic`` is the Z statistic with
+          pooled or unpooled variance, depending on the user choice of
+          `pooled` param. ``pvalue`` the probability of obtaining a test
+          statistic at least as extreme as the one that was actually
+          observed, assuming that the null hypothesis is true.
```

Comment:
This is over indented. Also not putting quotes here will render properly as a link to the object in the doc. And no need for object.
```suggestion
    ber : BarnardExactResult
       The object has attributes ``statistic`` and ``pvalue``.
       ``statistic`` is the Z statistic with
       pooled or unpooled variance, depending on the user choice of
       `pooled` param. ``pvalue`` the probability of obtaining a test
       statistic at least as extreme as the one that was actually
       observed, assuming that the null hypothesis is true.
```